### ISSUE TITLE
HTML Email

### ISSUE THREAD

**Issue by caseydm** [ubernostrum/django-registration#208]
I'm curious how others are implementing HTML activation emails. I found [this](https://stackoverflow.com/questions/1325983/django-send-email-in-html-with-django-registration) outdated Stack Overflow thread, but am not sure if it still applies.

**Comment by caseydm** [comment]
Ok disregard. I was able to do this by subclassing the send_activation_email method. In case anybody else wants to do this, here is how I did it:

```
# urls.py
url(r'^register/$', CustomRegistrationView.as_view(), name='register')

# views.py
from django.conf import settings
from django.core.mail import send_mail
from django.template.loader import render_to_string
from django.urls import reverse_lazy
from django_registration.backends.activation.views import RegistrationView

class CustomRegistrationView(RegistrationView):
    template_name = 'registration.html'
    form_class = CustomRegistrationForm
    email_subject_template = 'activation_email_subject.txt'
    success_url = reverse_lazy('activation_sent')

    def send_activation_email(self, user):
        """
        Custom method to enable HTML activation emails.
        """
        activation_key = self.get_activation_key(user)
        context = self.get_email_context(activation_key)
        context['user'] = user
        subject = render_to_string(
            template_name=self.email_subject_template,
            context=context,
            request=self.request
        )
        # Force subject to a single line to avoid header-injection
        # issues.
        subject = ''.join(subject.splitlines())
        text_content = render_to_string(
            template_name='activation_email_body.txt',
            context=context,
            request=self.request
        )
        html_content = render_to_string(
            template_name='activation_email_body.html',
            context=context,
            request=self.request
        )
        send_mail(
            subject,
            text_content,
            settings.DEFAULT_FROM_EMAIL,
            [user.email],
            html_message=html_content
        )
```

I used the activation email template [here](https://github.com/macropin/django-registration/blob/master/registration/templates/registration/activation_email.html) as a baseline:

```
{% load i18n %}
<!doctype html>
<html lang="en">

<head>
    <title>My Company Registration</title>
</head>

<body>
<p>
    You (or someone pretending to be you) have asked to register an account at
    My Company.  If this wasn't you, please ignore this email
    and your address will be removed from our records.
</p>
<p>
    To activate this account, please click the following link within the next
    7 days:
</p>

<p>
    <a href="{{scheme}}://{{site}}/activate_user/{{activation_key}}">
        {{scheme}}://{{site}}/activate_user/{{activation_key}}
    </a>
</p>
<p>
    Sincerely,<br>
    Company name
</p>
</body>

</html>
```

**Comment by mscheper** [comment]
@caseydm: The way I did it was to replace this library with [django-registration-redux](https://github.com/macropin/django-registration) (docs [here](https://django-registration-redux.readthedocs.io/en/latest/index.html)), which comes with HTML emails out of the box. Pretty much everything works out of the box, actually; I wish I'd seen it before I spent a day fiddling around with this legacy library, manually creating templates and getting them to work.

The good news is that replacing this library with the 'redux' version was pretty straightforward, and I haven't noticed any disadvantage. The only thing that wasn't documented obviously to me is that the reverse lookup for a logout URL is `{% url 'auth_logout' %}`.
